docs/frontend-system/building-plugins/migrating.md

```diff
@@ -0,0 +1,189 @@
+
+The main concept is that routes, components, apis are now extensions. You can use the appropriate extension creators to migrate all of them to extensions.
+
+## Pages
+
+Pages that were previously created using the `createRoutableExtension` extension function can be migrated to the new Frontend System using the `createPageExtension` extension creator, exported by `@backstage/frontend-plugin-api`.
+
+For example, given the following page:
+
+```ts
+export const FooPage = fooPlugin.provide(
+  createRoutableExtension({
+    name: 'FooPage',
+    component: () => import('./components').then(m => m.FooPage),
+    mountPoint: rootRouteRef,
+  }),
+);
+```
+
+it can be migrated as the following:
+
+```tsx
+import { createPageExtension } from '@backstage/frontend-plugin-api';
+import {
+  compatWrapper,
+  convertLegacyRouteRef,
+} from '@backstage/core-compat-api';
+
+const fooPage = createPageExtension({
+  defaultPath: '/foo',
+  // you can reuse the existing routeRef
+  // by wrapping into the convertLegacyRouteRef.
+  routeRef: convertLegacyRouteRef(rootRouteRef),
+  // these inputs usually match the props required by the component.
+  loader: ({ inputs }) =>
+    import('./components/').then(m =>
+      // The compatWrapper utility allows you to use the existing
+      // legacy frontend utilities used internally by the components.
+      compatWrapper(<m.FooPage />),
+    ),
+});
+```
+
+## Components
+
+The equivalent utility to replace components created with `createComponentExtension` is `createExtension` from `@backstage/frontend-plugin-api`. However, we recommend searching for an appropriate extension first, considering using `createExtension` as the last option for more complex scenarios.
+
+## APIs
+
+There are a few things to keep in mind in regards to utility APIs.
+
+### React package interface and ref changes
+
+Let's begin with [your `-react` package](../../architecture-decisions/adr011-plugin-package-structure.md). The act of exporting TypeScript interfaces and API refs have not changed from the old system. You can typically keep those as-is. For illustrative purposes, this is an example of an interface and its API ref:
+
+```tsx title="in @internal/plugin-example-react"
+import { createApiRef } from '@backstage/frontend-plugin-api';
+
+/**
+ * Performs some work.
+ * @public
+ */
+export interface WorkApi {
+  doWork(): Promise<void>;
+}
+
+/**
+ * The work interface for the Example plugin.
+ * @public
+ */
+export const workApiRef = createApiRef<WorkApi>({
+  id: 'plugin.example.work',
+});
+```
+
+In this example, the plugin ID already follows the [Frontend System Naming Patterns](../architecture/naming-patterns). If it doesn't, you may want to consider renaming that ID at this point. Don't worry, this won't hurt consumers in the old frontend system since the ID is mostly used for debugging purposes there. In the new system, it's much more important and appears in app-config files and similar.
```

Comment:
Oof, didn't spot this in the original review. This isn't correct, the ID is still very important and changing it will break existing usage. I think we could just remove this paragraph